api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -354,6 +354,9 @@ type FelixConfigurationSpec struct {
 	// flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the
 	// cluster.  It should not match the workload interfaces (usually named cali...).
 	BPFDataIfacePattern string `json:"bpfDataIfacePattern,omitempty" validate:"omitempty,regexp"`
+	// BPFL3IfacePattern is same as BPFDataIfacePattern but in addition it means that the device is a L3 tunnel device
```

Comment:
> This needs to match the interfaces that Calico workload traffic
￼	// flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the		￼	// flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the
￼	// cluster.  It should not match the workload interfaces (usually named cali...).		￼	// cluster.

Our tunnel interfaces clearly do not provide this. Hence we had to fix it :-)

So I'd say that

`BPFL3IfacePattern` is a regular expression that allows to list tunnel devices like wireguard or vxlan (i.e., L3 devices) in addition to `BPFDataIfacePattern`. That is, tunnel interfaces not created by Calico, that Calico workload traffic flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the cluster.